### PR TITLE
Feature/neat dpu names

### PR DESCRIPTION
**Pull request by eea02**
New requested feature is, that DPU names should be loaded from its own DPU resource bundle, like other localized messages. Also we should be able to distinguish from name displayed in UV in DPU tree menu and other places.
Loading names from DPU bundle should be optional, and thus configurable by application properties. Also there should be fall-back for situations, when localized names are missing in resource bundles.

### CODE REVIEW

File: commons-app/src/main/java/cz/cuni/mff/xrg/odcs/commons/app/module/DPUModuleManipulator.java [hunk 1]
```diff
@@ -32,6 +35,10 @@
     private static final Logger LOG = LoggerFactory
             .getLogger(DPUModuleManipulator.class);
 
+    public static final String DPU_NAME_KEY = "dpu.name";
+
+    public static final String DPU_MENU_NAME_KEY = "dpu.menu.name";
```
Comment (skrchnavy) [inline commons-app/src/main/java/cz/cuni/mff/xrg/odcs/commons/app/module/DPUModuleManipulator.java]: suggesterd to change to dpu.name.menu

**Comment by skrchnavy** [comment]
version of UV Core shall be 2.1.0 (currently snapshot). This is new feature.
It is also valid for DB scripts and updates

**Comment by skrchnavy** [comment]
@tomas-knap  please review, it is waiting 15 days

**Comment by tomas-knap** [comment]
Will review today

**Comment by tomas-knap** [comment]
The localized DPU names are stored to the database as dpuTemplate name/menuName, which is not correct. If I then switch localization of frontend, I see wrong dpu names, because these names are taken from database (dpuTemplate) and not from the resource bundles. Similarly, if i turn off/on dpu.name.localized config param, it does not behave correctly. 

Proposed solution: 
1)Localized versions of dpu name, dpu menu name should be always loaded dynamically from the resource bundles (EN/SK), never stored to the database. 
2) Dpu menu name is not saved to DB, so DB column menuName should be removed from DB table. 

Steps to reproduce
1) Set up dpu.name in resource bundle EN and SK. Build the DPU.
2) Set up locale of UnifiedViews (locale = en_US) in config.properties
3) Create new DPU template, use the one from 1)
4) Switch locale = sk in config.properties of UnifiedViews
5) Restart frontend
6) I would expect that I will see the proper SK label for DPU name, but this does not happen - I still see the wrong english label.

**Comment by skrchnavy** [comment]
As you have written, it is not good solution to load DPU names dynamicaly from the DPU. Due to the fact that we don't support per user localization but it is defined per installation, it shall be enough to define a mechanism to reload DPU templates via a command (via rest api or to have a button in Admin page of gui).

To run this always during startup is an overhead that is not necessary because it is not expected that locale is change in the system. One option would be to stre also locale in configuratuin table in DB and check if it differs from the one in config during start and run the 'refresh name' action only in case of change.

**Comment by eea04** [comment]
The cleanest and least error prone solution would be automatic reload at UV startup. The question is how long would such a reload take. If too long (>3s), some optimization might be reasonable. However, the "optimization" should definitely not require any user action apart from switching the locale itself, as that would be needlessly unsystematic and error prone.

**Comment by eea02** [comment]
Yes Tomas, this problem was known to me. I was assuming only case, that language and feature will be configured once, per UV installation. 
To support turning off this feature and language switching, I have added loading of DPU's at application start-up. UV already loads all DPU's at application start so performance will not be an issue. Now, if you decide to change language or turn-off feature, you will restart the application and info from bundle will be reloaded or default name will be used.
This leads to another known issue:
1. import DPU into UV
2. rename it
3. restart UV
Result: If DPU does not have parent, its name will be either taken from resource bundle or will be rewritten by default name.
This issue is acceptible, because for the future we plan to disable renaming of base DPU's. Renaming of child DPU's will be possible, and will not be affected by this issue.

**Comment by eea04** [comment]
I think that's not an 'issue', it's a desired limitation of the system.
The end users should not be able to change the base name of the DPU in the DPU tree. That's the responsibility of DPU designer. End users may change their preconfigured versions of the DPU and this is a functionality that remains.

**Comment by tomas-knap** [comment]
@eea02 Muzem zavolat kratce k objasneni toho problemu co pises? Skype "nappy_cz" a budu cca do jedne dnes.Dik

**Comment by tomas-knap** [comment]
Got the point. Ok, will discuss with CUNI team the impact of such commit - DPU names for 2nd level DPU templates cannot be customized, but it is taken either from pom.xml (in case of no localization message or switched off localization) or from resource bundles.

**Comment by tomas-knap** [comment]
CUNI is ok with this, just please @eea02 write explicitly what will happen when this feature will be installed on existing system with 2nd level DPU templates having custom names (with or without localization names turned on/off in config.properties). My expectation is that it will rewrite all the existing custom names so that they contain names either from pom.xml or from resource bundles (if localization names are turned on and there are proper localization messages). 

Will retest by tomorrow, so that we can finalize.

**Comment by eea02** [comment]
There are 3 possibilities what will happen depending on property file:
* dpu.name.localized = true - on startup, all DPU will be listed and installed. If name is found in its resource bundle, it will be rewritten, else name is leaved intact.
* dpu.name.localized = false - on startup, all DPU will be listed and installed. Its name will be rewritten by default name from manifest file.
* missing property 'dpu.name.localized' - will behave as dpu.name.localized = false

Maybe it would be benefical, if I alter this behaviour so if
* missing property 'dpu.name.localized' - on startup, all DPU will be listed and installed, its name will be leaven intact

**Comment by tomas-knap** [comment]
Will finish the testing in the morning - testing on my local instance causes problems (duplicate dpu templates after restart, dpu template not renamed after restart), needs to retest on fresh instance.

**Comment by tomas-knap** [comment]
1) set up SK localization, dpu.name.localized=true, fresh install, create new DPU template with localization messages. After restart and changing localization to EN -> OK
2) set up SK localization, dpu.name.localized=false, fresh install, create new DPU template with custom name. After restart, name is taken from pom.xml -> OK
3) set up SK localization, dpu.name.localized=true, fresh install, create new DPU template without localization messages. After restarting, name is untouched.

1), 2) works correctly, thanks. 
Regarding 3) When dpu.name.localized=true and localization message is not found in the proper resource bundle, currently it logs warning and does not change the name. I would suggest in this case to take name from pom.xml as fallback variant, so that wrong localization names are not shown in this case (e.g. after switching to EN variant, SK label should not be displayed if EN label is missing ) 

Otherwise OK.

**Comment by tomas-knap** [comment]
Regarding your proposal about missing dpu.name.localized (and 3 states of dpu.name.localized) - Lets keep it simple - if dpu.name.localized is missing, it is the same as if it would be set to false.

**Comment by skrchnavy** [comment]
@tomas-knap we have to cope with situation when 2nd level DPU templates were renamed, shall we revert renaming back to the name in manifest even user doesn't setup value for localization?
I support your decision but requires clear statement in release note that after upgrade of Core, names of DPU templates will be renamed.

**Comment by tomas-knap** [comment]
Yes, I discussed this with @jakubklimek and @skodape - Those DPUs which were renamed on the existing instances will have name changed to their pom.xml name if no localization message is provide or localization is disabled

**Comment by tomas-knap** [comment]
@eea02 Please adjust the "fallback variant" when localization is turned true and no localization message is provided (see above). Otherwise OK.

**Comment by eea02** [comment]
@tomas-knap fallback variant adjusted. Thanks. Closing feature.

**Comment by tomas-knap** [comment]
Notes: 
If "dpu.name.localized" in config.properties is set to false or missing, name/description is taken from pom.xml (see https://github.com/UnifiedViews/Core/pull/358) 

If  "dpu.name.localized" is set to true, localized version of the name is taken from resource bundle with localized messages (see https://github.com/UnifiedViews/Core/pull/358). Regarding description of the DPU template, see: https://github.com/UnifiedViews/Core/issues/387

**Comment by skrchnavy** [comment]
#84 is fixed by this PR

**Pull request merged**